doc/case-studies/timelock/README.md

```diff
@@ -0,0 +1,242 @@
+
+Again, imposing a simple structure which combines triggers with _or_, but effects with _and_ allows you to avoid cumbersome logic inside monitor predicates.
+
+Let's take a look at how reverse monitor specs help us to patch the loopholes described above:
+
+1. Overlooked side effects: a reverse monitor will detect _all_ changes to the monitored state, no matter where they originate.
+2. Side effects introduced during system evolution: same as above. Additionally, if an effect monitor is configured to revert in case of unexpected effects taking place, the developers will be forced to keep the spec in sync with the implementation.
+3. Inconsistent / spread-out specs: An effect monitor may describe all effects that may happen wrt. a particular state component in one place. As this monitor will be triggered when any of the methods that modify the state is executed, this also brings us considerable savings in terms of spec size/complexity, as similar effects can be grouped together.
+4. Unrestricted access from other contracts/methods: as in 1. and 2., it doesn't matter again where the modification comes from: if the state component we monitor is being changed, the monitor will detect it.
+
+
+## A use case: Timelock monitor specs
+
+Now it's time to look at the concrete example, the [Timelock contract][]. Go ahead, and compare:
+
+- Direct monitor specs in the [SCF 24 submission example][]
+- [Timelock monitor spec][] written in the style outlined here
+
+Let's go over the spec together here. (please note that this spec is not even a completely valid TLA+ yet, as e.g. many definitions need to be filled in from the contract code; this is tbd.)
+
+### Direct monitor for the `deposit` method
+
+The [deposit method](https://github.com/stellar/soroban-examples/blob/186443aab0bf8b7c2673428c38708bf38cb772ab/timelock/src/lib.rs#L57-L91) allows a user to send certain amount of token to this contracts, and to specify up to 10 claimants any of whom should be able to claim the funds when the specified timing condition fulfills. `MustFail` predicates in the spec closely follow the assertions in the code. Notice though that `MustFail_deposit_NotEnoughBalance` is not an assertion in the code, but a precondition which arises from usage of an external token contract. Nevertheless, the monitor system will force us to be complete (otherwise it will warn us about an unexpectedly reverted transaction).
+
+
+```tla 
+MustFail_deposit_TooManyClaimants == 
+    claimants.len > 10
+
+MustFail_deposit_Unauthorized == 
+    ~authorized(from)
+
+\* This property would not be originally conceived 
+\* if looking solely at the timelock contract source
+MustFail_deposit_NotEnoughBalance == 
+    token.balance(from) < amount
+
+\* Balance is externally relevant; initialized flag is not
+\* What matters for users is that balance is not overwritten
+MustFail_deposit_AlreadyInitialized == 
+    exists(balance)
```

Comment:
What is this `exists`? I can't seem to find a definition or reference.

Reply:
yeah, sorry, the defs are missing. `exists` is supposed to mean that the key is present in the ledger. We'll need to figure out the exact mapping between TLA+ and Soroban constructs